Client/SafeBrowsing/GSafeBrowsingDatabase.swift

```diff
@@ -315,6 +379,51 @@ class SafeBrowsingDatabase {
         })
     }
     
+    func scheduleUpdate(onNeedsUpdating: @escaping () -> Void) {
+        updateTimer?.invalidate()
+        updateTimer = Timer.scheduledTimer(withTimeInterval: self.lastFetchWaitDuration, repeats: false) {
+            $0.invalidate()
+            DispatchQueue.main.async {
+                onNeedsUpdating()
+            }
+        }
+    }
+    
+    private func calculateBackoffTime(_ numberOfRetries: Double) -> Double {
```

Comment:
It would be great to add a test for this because the algorithm is a little complicated.